worker/src/ee/experiments/experimentService.ts

```diff
@@ -0,0 +1,259 @@
+  input !== null &&
+  input !== undefined &&
+  !Array.isArray(input);
+
+const replaceVariablesInPrompt = (
+  prompt: PromptContent,
+  itemInput: Record<string, any>,
+  variables: string[],
+): { role: string; content: string }[] => {
+  const processContent = (content: string) =>
+    content.replace(/\{\{(\w+)\}\}/g, (_, variable) =>
+      variables.includes(variable)
+        ? (itemInput[variable] ?? "")
+        : `{{${variable}}}`,
+    );
+
+  if (typeof prompt === "string") {
+    return [{ role: ChatMessageRole.System, content: processContent(prompt) }];
+  } else {
+    return prompt.map((message) => ({
+      ...message,
+      content: processContent(message.content),
+    }));
+  }
+};
+
+const validateDatasetItem = (
+  itemInput: Prisma.JsonValue,
+  variables: string[],
+): itemInput is Prisma.JsonObject => {
+  if (!isValidPrismaJsonObject(itemInput)) {
+    return false;
+  }
+  return variables.some(
+    (variable) =>
+      Object.keys(itemInput).includes(variable) &&
+      typeof itemInput[variable] === "string",
+  );
+};
+
+export const createExperimentJob = async ({
+  event,
+}: {
+  event: z.infer<typeof ExperimentCreateEventSchema>;
+}) => {
+  logger.info("Processing experiment create job", event);
+  const { datasetId, projectId, runId } = event;
+
+  /********************
+   * INPUT VALIDATION *
+   ********************/
+
+  // first, fetch all the context required for the experiment
+  const datasetRun = await kyselyPrisma.$kysely
+    .selectFrom("dataset_runs")
+    .selectAll()
+    .where("id", "=", runId)
+    .where("project_id", "=", projectId)
+    .executeTakeFirstOrThrow();
+
+  if (!datasetRun.metadata) {
+    throw new ForbiddenError(
+      "Langfuse in-app experiments can only be run with available model and prompt configurations.",
+    );
+  }
+
+  const metadata = ExperimentMetadataSchema.safeParse(datasetRun.metadata);
+  if (!metadata.success) {
+    throw new ForbiddenError(
+      "Langfuse in-app experiments can only be run with available model and prompt configurations.",
+    );
+  }
+
+  // validate the prompt
+  const { prompt_id, provider, model, model_params } = metadata.data;
+
+  const prompt = await kyselyPrisma.$kysely
+    .selectFrom("prompts")
+    .selectAll()
+    .where("id", "=", prompt_id)
+    .where("project_id", "=", event.projectId)
+    .executeTakeFirstOrThrow();
+
+  if (!prompt) {
+    logger.error(`Prompt ${prompt_id} not found for project ${projectId}`);
+    throw new InvalidRequestError(
+      `Prompt ${prompt_id} not found for project ${projectId}`,
+    );
+  }
+
+  const validatePromptContent = PromptContentSchema.safeParse(prompt.prompt);
+
+  if (validatePromptContent.error) {
+    logger.error(
+      `Prompt content not in expected format ${prompt_id} not found for project ${projectId}`,
+    );
+    throw new InternalServerError(
+      `Prompt ${prompt_id} not found in expected format for project ${projectId}`,
+    );
+  }
```

Comment:
Out of curiosity: Any particular reason you checked !metadata.success in L98 and now condition on error being thruthy? IMO using !success makes sense in both cases.